src/map/Map.js

```diff
@@ -657,9 +657,23 @@ export var Map = Evented.extend({
 		var onResponse = Util.bind(this._handleGeolocationResponse, this),
 		    onError = Util.bind(this._handleGeolocationError, this);
 
-		if (options.watch) {
+		if (options.getCoordinates) {
+			if (options.watch) {
+				options.getCoordinates(options)
+					.then(function (locationWatchId) {
+						this._locationWatchId = locationWatchId;
+					});
+			} else {
+				options.getCoordinates(options)
+					.then(function (pos) {

```

Comment:
anonymous function not required,  I think you can just do this:
`options.getCoordinates(options).then(onResponse)`